### PR TITLE
R package

### PR DESCRIPTION
**Pull request by Dvermetten**
These are the changes to the code to make it into an R-package.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Package: IOHexperimenter
+Type: Package
+Title: Benchmarking Part of the 'IOHprofiler'
+Version: 0.0.0.9000
+Author: Furong Ye [cre, aut], Hao Wang [aut], Diederick Vermetten [aut], Carola Doerr [aut], Thomas Bäck [aut]
```
Comment (wangronin) [inline DESCRIPTION]: I would suggest putting you as the First author / creator

File: DESCRIPTION [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Maintainer: Furong Ye <f.ye@liacs.leidenuniv.nl>
+Description: The benchmarking module for the Iterative Optimization Heuristics
+    Profiler ('IOHprofiler'). This module provides benchmarking in the 'IOHprofiler'
+    format, which can be visulized using the 'IOHanalyzer' module.
+License: BSD_3_clause + file LICENSE
+Encoding: UTF-8
+LazyData: true
```
Comment (wangronin) [inline DESCRIPTION]: This means the lazy loading of the data set, if there is any, right?

File: DESCRIPTION [hunk 3]
```diff
@@ -0,0 +1,23 @@
+Package: IOHexperimenter
+Type: Package
+Title: Benchmarking Part of the 'IOHprofiler'
+Version: 0.0.0.9000
+Author: Furong Ye [cre, aut], Hao Wang [aut], Diederick Vermetten [aut], Carola Doerr [aut], Thomas Bäck [aut]
+Maintainer: Furong Ye <f.ye@liacs.leidenuniv.nl>
```
Comment (wangronin) [inline DESCRIPTION]: The same thing here: are two maintainer allowed? I think it is better to set you or me as the maintainer

File: R/IOHexperimenter.R [hunk 4]
```diff
@@ -1,68 +1,92 @@
...
+    paste0(functions, collapse = ','),
+    paste0(dims, collapse = ','),
+    paste0(instances, collapse = ',')
   )
-  
+
   # intialize the observer
   c_init_observer(
-    data.dir, algorithm_name, algorithm_info, 
+    data.dir, algorithm_name, algorithm_info,
```
Comment (wangronin) [inline R/IOHexperimenter.R]: Not exactly for this line.... Please ask Furong to verify the default value for parameters like number_interval_triggers, ....

File: R/IOHproblem.R [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+#' Reset the IOHproblem
+#'
+#' @param problem The IOHproblem object
+#'
+#' @return An IOHproblem object
+#' @export
+reset_problem <- function(problem) {
+  ans <- c_reset_problem()
```
Comment (wangronin) [inline R/IOHproblem.R]: Please check this with Furong: after calling c_reset_problem, what would happen to the output file? a new line in .info file? a new *.dat file?

File: src/interfaceR.cpp [hunk 6]
```diff
@@ -216,27 +218,35 @@ List reset_problem() {
...
-	}
+	// int N = x.size();
+	// int* x_ptr = (int*) malloc(sizeof(int) * N);
+	// for (int i = 0; i < N; ++i) {
+	// 	x_ptr[i] = x[i];
+	// }
 
-	IOHprofiler_evaluate_function(current_problem, x_ptr, result);
-	free(x_ptr);
+	IOHprofiler_evaluate_function(current_problem, INTEGER(x), result);
```
Comment (wangronin) [inline src/interfaceR.cpp]: INTEGER(x) will convert a IntergerVector type to int* ? I am not aware of this.

File: R/IOHproblem.R [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        if (is.null(dim(x))) x <- t(x)
+        if (ncol(x) != ans$dimension) x <- t(x)
+
+        stopifnot(ncol(x) == ans$dimension)
+        apply(x, 1, c_eval)
+      },
+      target_hit = function() {
+        c_is_target_hit()
+      },
+      set_parameters = function(param_vals){
```
Comment (wangronin) [inline R/IOHproblem.R]: This function sets the parameter we want to track, right? Do we also set the parameter to track by default for all problems, given a parameter list is provided when creating the IOHexperimenter class?

File: DESCRIPTION (hunk 1 above)
Comment (Dvermetten) [reply DESCRIPTION]: The official meaning of the [cre] tag seems to be:
the creator or maintainer, the person you should bother if you have problems.

It seems to be possible to put multiple creators, so should I just put you and Furong as creators as well?

File: DESCRIPTION (hunk 2 above)
Comment (Dvermetten) [reply DESCRIPTION]: Yes, this seems to be highly encouraged when creating a package

File: DESCRIPTION (hunk 3 above)
Comment (Dvermetten) [reply DESCRIPTION]: For the maintainer field, only one person is allowed. Should I put you or me there?

File: R/IOHproblem.R (hunk 7 above)
Comment (Dvermetten) [reply R/IOHproblem.R]: The way it is organized right now is as follows: when creating an IOHexperimenter class, you can specify which parameters to track. 
These can then be set in the IOHproblem classes derived from this. 
The default for this is no parameter tracking, and even if we were to set a default, the user would still need to manually call the set_parameter function in their code.

File: R/IOHproblem.R (hunk 5 above)
Comment (Dvermetten) [reply R/IOHproblem.R]: This reset_problem creates a new header-line in the *.dat-file and a new 'value' in the .info file. So until you go to the next function, you keep using the same .dat and .info files

File: src/interfaceR.cpp (hunk 6 above)
Comment (Dvermetten) [reply src/interfaceR.cpp]: Furong says this is indeed the case, and it seems to work correctly

File: R/IOHexperimenter.R (hunk 4 above)
Comment (Dvermetten) [reply R/IOHexperimenter.R]: I looked at the defaults with Furong, and they now match his original example configuration.

File: DESCRIPTION (hunk 3 above)
Comment (wangronin) [reply DESCRIPTION]: I think putting you would be good :)

File: R/IOHexperimenter.R (hunk 4 above)
Comment (wangronin) [reply R/IOHexperimenter.R]: Good! Thx!

File: R/IOHproblem.R (hunk 5 above)
Comment (wangronin) [reply R/IOHproblem.R]: Good, clear to me now!

File: src/interfaceR.cpp (hunk 6 above)
Comment (wangronin) [reply src/interfaceR.cpp]: Sure

File: DESCRIPTION (hunk 1 above)
Comment (wangronin) [reply DESCRIPTION]: Okay, I suggest the following:  Diederick Vermetten [cre, aut], Furong Ye [cre, aut], Hao Wang [aut], Carola Doerr [aut], Thomas Bäck [aut]

**Pull request merged**